miso-web/src/main/java/uk/ac/bbsrc/tgac/miso/webapp/context/WebConfig.java

```diff
@@ -70,4 +91,67 @@ public RequestMappingHandlerAdapter requestMappingHandlerAdapter() {
   public Map<ProgressStep.FactoryType, ProgressStepFactory> progressStepFactoryMap() {
     return progressStepFactories.stream().collect(Collectors.toMap(ProgressStepFactory::getFactoryType, Function.identity()));
   }
+
+  public class ExternalUriBuilder {
```

Comment:
Make static or extract from WebConfig into its own file. The latter is probably preferable to keep WebConfig clean